query/functor.hpp

```diff
@@ -652,11 +736,34 @@ struct BinaryFunctor<UUIDT, I> {
   }
 };
 
+template <typename I>
```

Comment:
Why we allow binary functor for geo point?

Reply:
I don't know. I didn't want to allow it.
the code won't compile without these specializations.